### PR TITLE
1.1/271 dupe message

### PR DESCRIPTION
**Pull request by katjam**

### CODE REVIEW

File: src/components/Queue.vue [hunk 1]
```diff
@@ -83,7 +83,7 @@ export default {
                 // so that we can reflect any changes in the Queue component.
                 var queueState = val.sendingStatus;
                 if(!queueState && val.sentData) {
-                    var message = this.genQueueSuccessMessage(val.sentData);
+                    var message = this.queueSuccessMessage(val.sentData);
```
Comment (ob6160) [inline src/components/Queue.vue:86]: Do we really need this method anymore if it's just returning val.sentData.message?

**Review by ob6160 — CHANGES_REQUESTED** [review]

File: src/components/Queue.vue (hunk 1 above)
Comment (katjam) [reply src/components/Queue.vue:86]: Nope but we are doing a refactor. So taking one step at a time. These are the final hours!

Comment (ob6160) [reply src/components/Queue.vue:86]: ok! :)

File: src/pages/Scan.vue [hunk 2]
```diff
@@ -128,17 +121,15 @@ export default {
...
                         if (!Store.netMgr.online) {
                             this.showQueued();
-                            this.setMessage("Not enough signal, voucher queued.", constants.MESSAGE_ERROR);
+                            this.setMessage("Not enough signal, voucher queued.", constants.MESSAGE_WARNING);
                         }
                     }.bind(this));
 
                 // Do anyway.
                 this.voucherCode = "";
-                this.sponsorCode = "";
```
Comment (ob6160) [inline src/pages/Scan.vue:137]: Why were these two lines removed? They're causing some of the tests to fail I think.

**Comment by ob6160** [comment]
This doesn't seem to be fully working - closing - will fix.

File: src/pages/Scan.vue (hunk 2 above)
Comment (katjam) [reply src/pages/Scan.vue:137]: Not sure. @alexalala Do you think that was a mistake?

File: src/pages/Scan.vue [hunk 3]
```diff
@@ -99,20 +100,12 @@ export default {
...
-                                this.setMessage("Please enter a valid voucher code.", constants.MESSAGE_ERROR);
-
-                            } else if (data.fail.length > 0) {
-                                this.showFail();
-                                this.setMessage("It looks like this code has already been added, please double check and try again. If you are still unable to add the voucher code, don\'t worry - you will still receive payment if you send it in with your other vouchers.", constants.MESSAGE_WARNING);
-                            }
-
+                        if (response.error) {
+                            this.showFail();
+                            this.setMessage(response.error, constants.MESSAGE_ERROR);
```
Comment (ob6160) [inline src/pages/Scan.vue:105]: Also the messages are part of the "response.data" object not the response itself.

File: src/pages/Scan.vue (hunk 2 above)
Comment (alexalala) [reply src/pages/Scan.vue:137]: @katjam @ob6160 quite possibly!!! Add them back in if they need to be :)

Comment (ob6160) [reply src/pages/Scan.vue:137]: They're pretty essential to the functioning of the scan page afaik.

File: src/pages/Scan.vue [hunk 4]
```diff
@@ -128,17 +121,15 @@ export default {
...
 
                 // Do anyway.
                 this.voucherCode = "";
-                this.sponsorCode = "";
-                this.$refs.sponsorBox.focus();
             } else {
-              this.showFail();
-              this.setMessage("Please enter a valid voucher code.", constants.MESSAGE_ERROR);
+                this.showFail();
+                this.setMessage(response.error, constants.MESSAGE_ERROR);
```
Comment (ob6160) [inline src/pages/Scan.vue:132]: There is no response in scope here.

File: src/pages/Scan.vue (hunk 2 above)
Comment (alexalala) [reply src/pages/Scan.vue:137]: @ob6160 yeah it was probably a mistake. As I said, add them back in if you need to

**Pull request closed without merging**